CMakeLists.txt

```diff
@@ -183,8 +174,9 @@ if(NOT (DJINNI_WITH_OBJC OR DJINNI_WITH_JNI OR DJINNI_WITH_PYTHON OR DJINNI_WITH
   message(FATAL_ERROR "At least one of DJINNI_WITH_OBJC or DJINNI_WITH_JNI or DJINNI_WITH_PYTHON or DJINNI_WITH_CPPCLI must be enabled.")
 endif()
 
+option(DJINNI_BUILD_TESTING "Build tests" ON)
 
 include(CTest)
-if (BUILD_TESTING)
+if (BUILD_TESTING AND DJINNI_BUILD_TESTING)
   add_subdirectory(test-suite)
 endif()
```

Comment:
Should we move the `include(CTest)` line inside the `if` block?

Reply:
Another option is also to always include the `test-suite` subdirectory and check if testing is enabled as the first thing on the CMakeLists.txt of that directory.

Reply:
The current configuration follows the [official suggestion how to use CTest](https://cmake.org/cmake/help/latest/module/CTest.html)

I am having a hard time explaining why it is like this. I am trusting the docs on this. All I know is that the value of `BUILD_TESTING` is defined as `ON` by default in `include(CTest)`

Reply:
> Another option is also to always include the test-suite subdirectory and check if testing is enabled as the first thing on the CMakeLists.txt of that directory.

I agree that this could be an option. I am having a hard time arguing against it, it just doesn't feel right. My only argument is that it wouldn't change a lot and not reduce complexity, but it would obfuscate the availability & impact of the `BUILD_TESTING` and `DJINNI_BUILD_TESTING` options if a user only reads the root `CMakeLists.txt`... 🤔